src/sentry/attachments/base.py

```diff
@@ -142,7 +146,10 @@ def get_data(self, attachment):
         data = []
 
         for key in attachment.chunk_keys:
-            data.append(zlib.decompress(self.inner.get(key, raw=True)))
+            raw_data = self.inner.get(key, raw=True)
+            if raw_data is None:
+                raise MissingAttachmentChunks()
```

Comment:
_nit_: Add the `key` into the message for "debugging"? It contains the project id, which could be useful in the future.

Reply:
This causes very poor grouping. I put the cache key into the wrapping log stmts instead, where we are able to intercept the formatting template

Reply:
You could do it by quoting it `"`, in which case we mask it for grouping.

Reply:
really? where is this happening

Reply:
Nvm, you're right. We only do it for hashes, but this won't be enough.

https://github.com/getsentry/sentry/blob/15ea025f431ee7981fdfeca6f7fc47322a32ccb9/src/sentry/grouping/strategies/message.py#L11

Reply:
It's a really cool idea though. I think we should consider building this